castle/cms/tiles/navigation.py

```diff
@@ -76,23 +78,79 @@ def items(self):
                     context = self.data_context
                 query = self.build_nav_query({
                     'path': {
-                        'depth': 1,
+                        'depth': depth,
                         'query': '/'.join(context.getPhysicalPath())
                     }
                 })
             if query is None:
                 return []
             limit = self.data.get('limit') or 10
             items = catalog(**query)[:limit + 40]
-            return [i for i in items if i.exclude_from_nav is not True][:limit]
+            brains = [i for i in items if i.exclude_from_nav is not True][:limit]
         else:
             uids = self.data.get('content') or []
             results = dict([(b.UID, b) for b in catalog(UID=uids)])
             items = []
             for uid in uids:
                 if uid in results:
                     items.append(results[uid])
-            return items
+            brains = items
+
+            if depth > 1:
+                paths = [x.getPath() for x in brains]
+                new_query = self.build_nav_query({
+                    'path': {
+                        'depth': depth,
+                        'query': paths
+                    }
+                })
+
+                brains = catalog(**new_query)
+
+        def find_parent(tree, child):
+
+            done = False
+            current = tree
+
+            while not done:
+                # We're basically just walking a directory tree
+                # as long as the paths match, keep going deeper until
+                # the item we're placing is in the correct place
+
+                found = False
+                for item in current['sub_items']:
+                    if item['path'] in child[0]:
+                        current = item
+                        found = True
+
+                        # need to break, or we'll keep looking at the current
+                        # level
+                        break
+
+                if not found:
+                    # there's no objects at this level in the tree
+                    # that "child" can fit into, so place it along side
+                    done = True
+
+            current['sub_items'].append({
+                'path': child[0],
+                'obj': child[1],
+                'sub_items': []
+            })
+
+            return tree
+
+        brain_listing = [[x.getPath(), x] for x in brains]
+        sorted_brain_listing = sorted(brain_listing, key=lambda x: x[0])
+
+        limit = self.data.get('limit') or 10
```

Comment:
Eh? The "limit" setting is a required option, so it should ideally never need to use that